kratos/future/solving_strategies/schemes/implicit_scheme.h

```diff
@@ -0,0 +1,1772 @@
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief unction to be called when it is needed to initialize an iteration. It is designed to be called at the beginning of each non linear iteration
+     * @note Take care: the elemental function with the same name is NOT called here.
+     * @warning Must be defined in derived classes
+     * @details The function is called in the builder for memory efficiency
+     * @param rModelPart The model part of the problem to solve
+     * @param rLinearSystemContainer Auxiliary container with the linear system arrays
+     */
+    virtual void InitializeNonLinIteration(LinearSystemContainer<TSparseMatrixType, TSystemVectorType> &rLinearSystemContainer)
+    {
+        KRATOS_TRY
+
+        // Finalizes non-linear iteration for all of the elements, conditions and constraints
+        EntitiesUtilities::InitializeNonLinearIterationAllEntities(*mpModelPart);
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief Function to be called when it is needed to finalize an iteration. It is designed to be called at the end of each non linear iteration
+     * @param rModelPart The model part of the problem to solve
+     * @param rLinearSystemContainer Auxiliary container with the linear system arrays
+     */
+    virtual void FinalizeNonLinIteration(LinearSystemContainer<TSparseMatrixType, TSystemVectorType> &rLinearSystemContainer)
+    {
+        KRATOS_TRY
+
+        // Finalizes non-linear iteration for all of the elements, conditions and constraints
+        EntitiesUtilities::FinalizeNonLinearIterationAllEntities(*mpModelPart);
+
+        KRATOS_CATCH("")
+    }
+
+    /**
+     * @brief Set the Up Dof Arrays
+     * This method sets the standard and effective DOF sets
+     * @param pDofSet Pointer to the standard DOF set
+     * @param pEffectiveDofSet Pointer to the effective DOF set
+     * @param rSlaveToMasterDofsMap The map containing the corresponding master(s) for each slave DOF
+     * @return std::pair<std::size_t, std::size_t> Sizes of the standard and effective DOF sets
+     */
+    virtual std::pair<std::size_t, std::size_t> SetUpDofArrays(
+        typename DofsArrayType::Pointer pDofSet,
+        typename DofsArrayType::Pointer pEffectiveDofSet,
+        DofArrayUtilities::SlaveToMasterDofsMap& rSlaveToMasterDofsMap)
+    {
+        // Call the external utility to set up the DOFs array
+        DofArrayUtilities::SetUpDofArray(*mpModelPart, *pDofSet, mEchoLevel);
+        KRATOS_INFO_IF("ImplicitScheme", mEchoLevel >= 2) << "Finished DOFs array set up." << std::endl;
+
+        // Call the external utility to set up the DOFs array
+        DofArrayUtilities::SetUpEffectiveDofArray(*mpModelPart, *pDofSet, *pEffectiveDofSet, rSlaveToMasterDofsMap, mEchoLevel);
+        KRATOS_INFO_IF("ImplicitScheme", mEchoLevel >= 2) << "Finished effective DOFs array set up." << std::endl;
+
+        // Set the corresponding flag
+        mDofSetIsInitialized = true;
```

Comment:
@KratosMultiphysics/technical-committee check if we need this one. If not remove it.

Reply:
Done.